android/app/src/main/java/com/sample/MainActivity.kt

```diff
@@ -32,21 +35,67 @@ class MainActivity : ReactActivity() {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
+        val reactNativeHost = reactNativeHost
+        if (reactNativeHost is TestAppReactNativeHost) {
+            reactNativeHost.currentActivity = this
+        }
+
         val manifest = manifestProvider.manifest
-            ?: throw IllegalStateException("app.json is not provided or TestApp is misconfigured")
+                ?: throw IllegalStateException("app.json is not provided or TestApp is misconfigured")
+        findViewById<MaterialToolbar>(R.id.topAppBar).apply {
```

Comment:
Nit: it would be great to extract this call into own `setupToolbar(displayName: String)` method to declutter the `onCreate()` a little bit. The same applies to the recyclerView setup. I can do that, just let me know if you're okay with it :)

Reply:
Let's do that in a separate PR.